app/routes/course-admin/code-example-insights.ts

```diff
@@ -0,0 +1,44 @@
+import { inject as service } from '@ember/service';
+import BaseRoute from 'codecrafters-frontend/utils/base-route';
+import Store from '@ember-data/store';
+import type CourseStageModel from 'codecrafters-frontend/models/course-stage';
+import type LanguageModel from 'codecrafters-frontend/models/language';
+import type CommunityCourseStageSolutionModel from 'codecrafters-frontend/models/community-course-stage-solution';
+
+export type CodeExampleInsightsRouteModel = {
+  courseStage: CourseStageModel;
+  language: LanguageModel;
+  solutions: CommunityCourseStageSolutionModel[];
+};
+
+export default class CodeExampleInsightsRoute extends BaseRoute {
+  @service declare store: Store;
+
+  queryParams = {
+    language_slug: {
+      refreshModel: true,
+    },
+    sort_mode: {
+      refreshModel: true,
+    },
+  };
+
+  async model(params: { stage_slug: string; language_slug?: string; sort_mode?: string }): Promise<CodeExampleInsightsRouteModel> {
+    // @ts-ignore
+    const course = this.modelFor('course-admin').course as CourseModel;
+    const courseStage = course.stages.findBy('slug', params.stage_slug);
+    const languages = course.betaOrLiveLanguages;
+    const selectedLanguage = languages.findBy('slug', params.language_slug);
+
+    return {
+      courseStage: courseStage,
+      language: selectedLanguage,
```

Comment:
Check why this isn't caught by types - if language could be null and the model says not-null this should've triggered a failure